### PR TITLE
Docs: Distinguish examples in rules under Best Practices part 1

### PR DESCRIPTION
**Pull request by pedrottimark**
1 of 4 pull requests for rules under **Best Practices** Have a good week end!

This gets us to 25% of the rules. We might finish in about 4 weeks and celebrate April Rules day.

What do you think about the wording of these example sentences?
* complexity: Examples of incorrect/correct code for a maximum of 2
* dot-notation: Examples of correct code for the sample `{ "allowPattern": "^[a-z]+(_[a-z]+)+$" }` option

Here are the changes beyond example sentences:

* accessor-pairs: Options unordered list with default option first; delete json
* array-callback-return: add eslint comments to example code
* block-scoped-var: delete redundant code in introduction; rename functions to merge code fences in examples; add third function to correct code
* complexity: add Options heading
* curly: move default example under all option; delete redundant json for options
* dot-location: unordered list for options; delete json; simplify description of options
* dot-notation: under Options, replace json with unordered list whose items are sentences which were under the option headings
* eqeqeq: delete json; put examples under options in incorrect/correct order; add to list of rules with *missing* correct code; *tied for the cut-and-paste prize in this batch*
* no-case-declarations: delete redundant incorrect intro example; move correct code example from intro to Rule Details; add eslint-env comments; *tied for the cut-and-paste prize in this batch*

WHAT DO YOU THINK about potential change to array-callback-return: move Note to **Known Limitations** (which would precede When Not To Use It)

This rule checks callback functions of methods with the given names, *even if* the object which has the method is *not* an array.

### CODE REVIEW

File: docs/rules/complexity.md [hunk 1]
```diff
@@ -57,7 +59,7 @@ Optionally, you may specify a `maximum` object property:
 is equivalent to
 
 ```json
-"complexity": [2, {"maximum": 2}]
+"complexity": [2, { "maximum": 2} ]
```
Comment (platinumazure) [inline docs/rules/complexity.md]: I see a space after opening brace, no space before closing brace, and space before closing square bracket (with no space after opening square bracket). Could you make this more consistent please?

Comment (pedrottimark) [inline docs/rules/complexity.md]: :+1: my bad

**Comment by nzakas** [comment]
> WHAT DO YOU THINK about potential change to array-callback-return: move Note to Known Limitations (which would precede When Not To Use It)

:+1:

**Comment by pedrottimark** [comment]
* array-callback-return: moved Note to Known Limitations
* complexity: corrected misplaced space

**Comment by ilyavolodin** [comment]
LGTM! Thanks a lot for doing this!

**Pull request merged**